### DISCUSSION TITLE
Cannot generate plots in RStudio

### DISCUSSION THREAD

**Discussion by kaparr94** [ImmuneDynamics/Spectre#112] [Help] [answered]
Hello All,

I'm new to RStudio (and programming in general). I cannot generate plots after applying arcsinh transformation within the simple discovery workflow. 

I input the command exactly as worded (copy and paste) from the protocol page for simple discovery workflow, which is:

for(i in transformed.cols){
  make.colour.plot(do.subsample(cell.dat, 20000), i, plot.against)
}

Here is the error message that pops up:
Loading required package: ggplot2
Error: package or namespace load failed for ‘ggplot2’ in loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]]):
 object 'pkgInfo' not found
Loading required package: scales
Error: package or namespace load failed for ‘scales’ in loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]]):
 object 'pkgInfo' not found
Loading required package: ggthemes
Error: package or namespace load failed for ‘ggthemes’ in loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]]):
 object 'pkgInfo' not found
Error in ggplot(data = dat, aes(x = .data[[x.axis]], y = .data[[y.axis]])) : 
  could not find function "ggplot"
In addition: Warning messages:
1: package ‘ggplot2’ was built under R version 4.0.5 
2: In loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]]) :
  package ‘rlang’ has no 'package.rds' in Meta/
3: package ‘scales’ was built under R version 4.0.5 
4: In loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]]) :
  package ‘rlang’ has no 'package.rds' in Meta/
5: package ‘ggthemes’ was built under R version 4.0.5 
6: In loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]]) :
  package ‘rlang’ has no 'package.rds' in Meta/

Much appreciated!

-Kayla

**Comment by kaparr94** [comment]
I've fixed the issue. If you want to know how, comment and I'll explain.

**Reply by tomashhurst** [reply]
@kaparr94 if it's OK, I would love to know! Might help us down the road if someone comes across a similar issue.

**Reply by kaparr94** [reply]
Yes, of course! I just posted the answer below. :)

**Comment by kaparr94** [comment, marked as answer]
The error messages shown in my original post were being caused by the location of the packages within my computer. I had R (including all its subfolders and thus the libraries) saved to my OneDrive, which aren't technically located on my local computer. R packages cannot load unless they're saved to your local C disk and this will cause the "pkginfo not found" error.

To fix: I copied the entire R folder from its location within my OneDrive folder and pasted it to my C drive. For me, the new path became "C:/Users/kp299/Documents/R/R-4.0.3/packages"

In order to update this new path so that RStudio knows where to look for packages when loading them, I used the command .libPaths("C:/Users/kp299/Documents/R/R-4.0.3/packages")

Of course, you'd want to place your own path within the ("").

You can check whether the path was accepted by typing the command .libPaths
and it should return with your updated path.

For whatever reason though, you have to repeat this .libPaths command to update the path after each new session of RStudio or else RStudio will give you the error of "package not found" when loading packages. Not sure how to permanently fix but it works for me.
